deeplake/util/keys.py

```diff
@@ -183,15 +184,36 @@ def get_sequence_encoder_key(key: str, commit_id: str) -> str:
 
 
 def dataset_exists(storage) -> bool:
+    """
+    Returns true if a dataset exists at the given location.
+    NOTE: This does not verify if it is a VALID dataset, only that it exists and is likely a deeplake directory.
+    To verify the content, use :func:`dataset_valid`
+    """
     try:
-        storage[get_dataset_meta_key(FIRST_COMMIT_ID)]
-        return True
+        return (
+                get_dataset_meta_key(FIRST_COMMIT_ID) in storage
+                or get_version_control_info_key() in storage
+        )
     except S3GetAccessError as err:
         raise AuthorizationException("The dataset storage cannot be accessed") from err
-    except (KeyError, S3GetError) as err:
+    except S3GetError:
```

Comment:
what if it is in GCS, would this error catch it or is it only for managed datasets?

Reply:
There wasn't any exceptions beyond S3 ones there before, so I just left that as it was except for the KeyError which I **_think_** was only triggered previously when we'd look up a value in the dict interface which doesn't exist and now that it's doing an `in` I didn't think that exception would be thrown.

BUT: I did think it was odd how only S3 had bubbled up exceptions of all the different providers. Maybe GCS and Azure just throw KeyError exceptions instead of something more specialized? I can certainly put the KeyError back in. Or, better yet, if we run tests against those environments we'd know for sure what they do.

Reply:
@farizrahman4u or @FayazRahman do you have thoughts here?

Reply:
I'm putting the KeyError back in with the next commit, just to be safe/flexible.